Microsoft.Azure.CosmosRepository/src/Pagination/IPagedResultRequest.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.CosmosRepository.Pagination
+{
+    /// <summary>
+    /// This interface is defined to standardize to request a paged result.
+    /// </summary>
+    public interface IPagedResultRequest : ILimitedResultRequest
```

Comment:
The interface inheritance seems to mean that you don't allow skipping (OFFSET) without also taking (LIMIT), although Cosmos does seem to support that. For example, it would be possible to express "give me all elements, skipping the first 100".

Reply:
Thank you for the review, the current design does not support the scenario, we can support it easily in the future, we just add another interface `ISkipResultRequest`, and does not inherit the `ILimitedResultRequest`, then add a new branch in the below method. In my opinion, most of the scenarios need the max result count, any suggestions?
```
protected virtual IQueryable<TItem> ApplyPaging<TPagedRequest>(IQueryable<TItem> query, TPagedRequest pagedResultRequest)
{
        //Try to use paging if available
        if (pagedResultRequest is IPagedResultRequest pagedInput)
        {
            return query.Skip(pagedInput.SkipCount).Take(pagedInput.MaxResultCount);
        }

        //Try to limit query result if available
        if (pagedResultRequest is ILimitedResultRequest limitedInput)
        {
            return query.Take(limitedInput.MaxResultCount);
        }

        //Try to limit query result if available
        if (pagedResultRequest is ISkipedResultRequest skipInput)
        {
            return query.Skip(skipInput));
        }

        //No paging
        return query;
}
```

Reply:
I think @roji is on to something here, perhaps we should consider this a bit further. I want to make sure this is right. Also, I was informed that the Cosmos DB team actually suggestions continuation tokens instead of the Skip / Take approach to pagination.

Reply:
Hi @IEvangelist, I added some code to support the scenario accordingly, now we can skip the specific count and get all the items.
About the continuation tokens, I noticed the document from the Cosmos DB team, the solution is better than the skip/take the approach, I need to more effort on it since I don't familiar with the knowledge, can we merge the PR first, then we can add enhancement to support the continuation tokens in the future.

Reply:
That's the thing, I'm not sure the API will be the same. I'm hesitant to introduce a new API that may not be correct and might change with the next PR. I'd rather do it right the first time. I think what we have is awesome, but I want to explore the continuation approach first.

Reply:
@IEvangelist  I totally agree with what you said doing it right the first time. I will close the PR and continue to explore the continuation approach in the future. currently, I will use the branch DLL in my project first, I can replace the branch DLL with official NuGet if the continuation approach is available in the future, thank you to discuss with me about the feature.

Reply:
I really appreciate it, thank you. Sorry to push back, but I just want to make sure it's the right solution since more people are starting to use this library. In fact, would you be interested in helping maintain it with me? Now that I'm starting to write a book, I have less time to give this project the full attention it deserves. I'd love 💜 more help.

Reply:
@IEvangelist I am glad you can invite me to maintain the project with you, but I am afraid I can not take the responsibility since I am a newbie in Cosmos DB, however, I will pay attention to the library and contribute my efforts if I can, thank you again.